pkg/virt-controller/watch/vmi_test.go

```diff
@@ -3125,6 +3130,80 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 						"[{\"interface\":\"net1\",\"name\":\"net1\",\"namespace\":\"default\"},{\"interface\":\"net2\",\"name\":\"net1\",\"namespace\":\"default\"}]")),
 			)
 		})
+
+		Context("interface status", func() {
+			const (
+				ifaceName   = "iface1"
+				networkName = "meganet"
+			)
+
+			DescribeTable("hotplugged interface", func(vmi *virtv1.VirtualMachineInstance, expectedIfaces ...virtv1.Interface) {
+				Expect(hotpluggedInterfaces(vmi)).To(ConsistOf(expectedIfaces))
+			},
+				Entry("VMI without interfaces on vmi.Spec has no interfaces to hot-plug", api.NewMinimalVMI(vmName)),
+				Entry("VMI with interface on vmi.Spec, also reflected on vmi.Status.Interfaces does not have ifaces to hot-plug",
+					newVMIWithOneIfaceStatus(newVMIWithOneIface(api.NewMinimalVMI(vmName), networkName, ifaceName), ifaceName)),
+				Entry("VMI with interface on vmi.Spec not found on vmi.Status.Interfaces does not have ifaces to hot-plug",
+					newVMIWithOneIface(api.NewMinimalVMI(vmName), networkName, ifaceName),
+					virtv1.Interface{
+						Name: ifaceName,
+					}),
+			)
+
+			DescribeTable("updateInterfaceStatus", func(vmi *virtv1.VirtualMachineInstance, ifaceStatus ...PodVmIfaceStatus) {
+				var (
+					podIfaceStatus []networkv1.NetworkStatus
+					vmIfaceStatus  []virtv1.VirtualMachineInstanceNetworkInterface
+				)
+				for i, ifaceState := range ifaceStatus {
+					if ifaceState.podIfaceStatus != nil {
+						podIfaceStatus = append(podIfaceStatus, *ifaceStatus[i].podIfaceStatus)
+					}
+					if ifaceState.vmIfaceStatus != nil {
+						vmIfaceStatus = append(vmIfaceStatus, *ifaceStatus[i].vmIfaceStatus)
+					}
+				}
+
+				Expect(controller.updateInterfaceStatus(vmi, NewPodForVirtualMachine(vmi, k8sv1.PodRunning, podIfaceStatus...))).To(Succeed())
+				Expect(vmi.Status.Interfaces).To(ConsistOf(vmIfaceStatus))
+			},
+				Entry("VMI without interfaces on spec does not generate new interface status", api.NewMinimalVMI(vmName)),
+				Entry("VMI with an interface on spec (not matched on status) generates new interface status",
+					newVMIWithOneIface(api.NewMinimalVMI(vmName), networkName, ifaceName),
+					PodVmIfaceStatus{
+						vmIfaceStatus: &v1.VirtualMachineInstanceNetworkInterface{Name: ifaceName},
+					}),
+				Entry("VMI with an interface on spec (matched on status) does not generate new interface status",
+					newVMIWithOneIfaceStatus(newVMIWithOneIface(api.NewMinimalVMI(vmName), networkName, ifaceName), ifaceName),
+					PodVmIfaceStatus{
+						vmIfaceStatus: simpleIfaceStatus(ifaceName),
+					}),
+				Entry("VMI with an interface on spec - *not* matched on status - is an hotplug request",
```

Comment:
This test seems to have the wrong title and to be identical to the one after it - `VMI with an interface on spec (matched on status) but with the pod interface not ready`.

Reply:
Done! Amazing catch. 

I've pruned the redundant tests - I think this is the result of a rebase gone bad (I had the same function w/ two different names, and ended up duplicating everything - including the tests).